lib/utils/resolveObject.js

```diff
@@ -100,9 +97,24 @@ function compile_value(value, stack) {
         // Recursive so we can compute multi-layer variables like a = b, b = c, so a = c
         if ( to_ret.indexOf('{') > -1 ) {
           var reference = to_ret.slice(1, -1);
-          if(stack.indexOf(reference)!==-1) {
-            stack.push(reference);
-            throw new Error('Variable reference for \'' + stack[0] + '\' resolves to circular definition cycle of \'' + reference + '\':  ' + stack.join(', ') );
+
+          // Compare to found circular references
+          var foundCircArr = Object.keys(foundCirc);
+          if (foundCircArr.indexOf(reference)!==-1) {
+            // If the current reference is a member of a circular reference, do nothing
+          }
+          else if(stack.indexOf(reference)!==-1) {
```

Comment:
It's weird to have an empty body to part of a conditional, this could be rewritten as

```js
if (stack.indexOf(reference) !== -1) {
} else if (foundCircArr.indexOf(reference) === -1) {
}
```

Reply:
Yeah. We may want to do something in the future. I think we had something there before but I removed it. I like having the comment in there to be clear that we are intentionally not doing something there.

Alternatively, I could refactor the logic. Let me know if you think that would be better.